Tools/autotest/sim_vehicle.py

```diff
@@ -570,6 +570,10 @@ def start_vehicle(binary, autotest, opts, stuff, loc):
         paths = [os.path.join(autotest, x) for x in paths]
         path = ",".join(paths)
         progress("Using defaults from (%s)" % (path,))
+        if opts.user_param_file:
```

Comment:
This will only reach here if there is a param file to be loaded, which I think is wrong. It should be something like:
```
paths = []
if "default_params_filename" in stuff:
    paths = stuff["default_params_filename"]
    if not isinstance(paths, list):
        paths = [paths]
    paths = [os.path.join(autotest, x) for x in paths]
if opts.user_param_file:
    paths += str(opts.user_param_file)    // too late to remember if this works to add stuff to a dict
path = ",".join(paths)
progress("Using defaults from (%s)" % (path,))
```